src/collision.c

```diff
@@ -0,0 +1,1173 @@
+#include "common.h"
+#include "model.h"
+
+extern s16 D_800D91DC;
+
+typedef struct {
+    u32 collisionOffset;
+    u32 zoneOffset;
+} HitAsset;
+
+typedef struct {
+    s32 flags;
+    s16 parentModelIndex;
+    s16 unk_06;
+} ColliderBackupEntry;
+
+typedef struct {
+    s16 numColliders;
+    s16 unk_02;
+    s32 collidersOffset;
+    s16 numVertices;
+    s16 unk_0a;
+    s32 verticesOffset;
+    s16 boundingBoxesDataSize;
+    s16 unk_12;
+    s32 boundingBoxesOffset;
+} HitAssetCollisionData;
+
+typedef struct {
+    s16 boundingBoxOffset;
+    s16 nextSibling;
+    s16 firstChild;
+    s16 numTriangles;
+    s32 trianglesOffset;
+} HitAssetCollider;
+
+extern ColliderBackupEntry* gCollisionDataBackup;
+extern ColliderBackupEntry* gCollisionDataZoneBackup;
+extern CollisionData gZoneCollisionData;
+
+extern f32 gCollisionRayStartX;
+extern f32 gCollisionRayStartY;
+extern f32 gCollisionRayStartZ;
+extern f32 gCollisionRayDirX;
+extern f32 gCollisionRayDirY;
+extern f32 gCollisionRayDirZ;
+extern f32 gCollisionPointX;
+extern f32 gCollisionPointY;
+extern f32 gCollisionPointZ;
+extern f32 gCollisionRayLength;
+extern f32 gCollisionNormalX;
+extern f32 gCollisionNormalY;
+extern f32 gCollisionNormalZ;
+
+s32 collision_heap_create(void);
+void* collision_heap_malloc(s32 size);
+void collision_heap_free(void*);
+
+void load_hit_data(s32 idx, HitAsset* hit);
+void _add_hit_vert_to_buffer(Vec3f** buf, Vec3f* vert, s32* bufSize);
+s32 _get_hit_vert_index_from_buffer(Vec3f** buffer, Vec3f* vert, s32* bufferSize);
+
+void backup_map_collision_data(void)
+{
```

Comment:
Hia, we use this brace style:
```suggestion
void backup_map_collision_data(void) {
```